genesyscloud/resource_genesyscloud_integration.go

```diff
@@ -0,0 +1,495 @@
+	integration, _, err := integrationAPI.PostIntegrations(createIntegration)
+
+	if err != nil {
+		return diag.Errorf("Failed to create integration %s: %s", name, err)
+	}
+
+	d.SetId(*integration.Id)
+
+	// Set attributes that can only be modified in a patch
+	if d.HasChange(
+		"intended_state") {
+		log.Printf("Updating additional attributes for integration %s", name)
+		_, _, patchErr := integrationAPI.PatchIntegration(d.Id(), platformclientv2.Integration{
+			IntendedState: &intendedState,
+		}, 25, 1, "", nil, "", "")
+
+		if patchErr != nil {
+			return diag.Errorf("Failed to update integration %s: %v", name, patchErr)
+		}
+	}
+
+	//Update integration config separately
+	diagErr := updateIntegrationConfig(d, integrationAPI)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Printf("Created integration %s %s", name, *integration.Id)
+	return readIntegration(ctx, d, meta)
+}
+
+func readIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	integrationAPI := platformclientv2.NewIntegrationsApiWithConfig(sdkConfig)
+
+	log.Printf("Reading integration %s", d.Id())
+	currentIntegration, resp, getErr := integrationAPI.GetIntegration(d.Id(), 100, 1, "", nil, "", "")
+
+	if getErr != nil {
+		if isStatus404(resp) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Failed to read integration %s: %s", d.Id(), getErr)
+	}
+
+	d.Set("name", *currentIntegration.Name)
+	if currentIntegration.IntendedState != nil {
+		d.Set("intended_state", *currentIntegration.IntendedState)
+	} else {
+		d.Set("intended_state", nil)
+	}
+
+	// Use returned ID to get current config, which contains complete configuration
+	integrationConfig, _, err := integrationAPI.GetIntegrationConfigCurrent(*currentIntegration.Id)
+
+	if err != nil {
+		return diag.Errorf("Failed to read config of integration %s: %s", d.Id(), getErr)
+	}
+
+	d.Set("config", flattenIntegrationConfig(integrationConfig))
+
+	// Use integration type Id to get complete type object
+	typeName := *currentIntegration.IntegrationType.Id
+	integrationType, _, err := integrationAPI.GetIntegrationsType(typeName)
+
+	if err != nil {
+		return diag.Errorf("Failed to read integration type of integration %s: %s", d.Id(), getErr)
+	}
+
+	d.Set("integration_type", flattenIntegrationType(integrationType))
+
+	log.Printf("Read integration %s %s", d.Id(), *currentIntegration.Name)
+
+	return nil
+}
+
+func updateIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	name := d.Get("name").(string)
+	intendedState := d.Get("intended_state").(string)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	integrationAPI := platformclientv2.NewIntegrationsApiWithConfig(sdkConfig)
+
+	if d.HasChange("intended_state") {
+
+		log.Printf("Updating integration %s", name)
+
+		diagErr := retryWhen(isVersionMismatch, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
```

Comment:
Do the integrations potentially return version mismatch errors? The directory APIs (users, skills, etc.) have this issue due to their database schema, but I would assume integrations are completely different and wouldn't return the same error.